cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RuntimeJobTwillPreparer.java

```diff
@@ -133,49 +129,7 @@ private RuntimeJobInfo createRuntimeJobInfo(RuntimeSpecification runtimeSpec,
       }
     }
 
-    return new DefaultRuntimeJobInfo(getProgramRunId(), resultingFiles, jvmProperties);
+    return new DefaultRuntimeJobInfo(getProgramRunId(), resultingFiles, jvmOpts);
   }
 
-  /**
-   * Parses a jvm options string (e.g. {@code -XYZ -Dkey=value}) and extracts key/value properties
-   * provided by the {@code -D} options.
-   */
-  @VisibleForTesting
-  static Map<String, String> parseJvmProperties(String jvmOpts) {
```

Comment:
why do we need to move this code from here to the RuntimeJob?

the reason I ask is that this class is used on the instance side, before the job is submitted, whereas RuntimeJob is executed on the cluster. So if there is an invalid value it would be better to fail earlier.

Reply:
ah misread the classname